docs/NewBackendSpecificNode.md

```diff
@@ -0,0 +1,96 @@
+
+Different layout leads to different implementation of the operator kernel on certain backend. 
+For example, in the subgraph from ResNet50, a `CPUConvDKKC8` node with memory layout modified for efficient SIMD access is introduced to optimized for CPU backend. Please refer to "5.3 Use Case: Optimizing Resnet50 for the CPU" in [Glow paper](https://arxiv.org/abs/1805.00907).
+
+We should take both fast operator kernel implementation and extra potential layout transformation into consideration to get better performance.
+
+### Steps
+
+This section is more accurately about (1) adding new backend-specific nodes and corresponding instructions, (2) how to utilize the APIs to add these backend-specifics nodes to the graph, and (3) how to have your backend correctly handle this new corresponding backend-specific instruction that is IRGen'd from your backend-specific Node.
+
+Here are mainly three steps to add a new backend-specific node in Glow:
+
+1. Add a backend-specific Node `FusedAB` to `XSpecificNodes.h`, and a corresponding backend-specific Instruction `FusedAB` to `XSpecificInstrs.h`. Note that the `FusedABInst` needs to be marked with `autoIRGen()` so that the node is automatically IRGen'd to the instruction, as we currently do not support backend-specific IRGen.
+
+2. Add logic to `XBackend::transformPreLowering()` or `XBackend::transformPostLowering()` (or both) depending on when you want to do the transformation. This logic would look for the pattern of nodes you want to fuse (`A` with a single use by `B`), and replaces all uses of the result of B with the new backend-specific `FusedABNode`.
+
+3. Have your backend `X` implement `FusedABInst`. For example, for the OpenCL backend, this would mean adding a case to enqueue a kernel for the `FusedABInst` to `OpenCLFunction::execute()`, and then adding the corresponding kernel in `kernels.cl`.
+
+
+### Examples
+
+#### Operator Fusion for ReLU in CPU
+
+ReLU is max between zero and the input value. Glow lowers `ReLUNode` to two basic low-level linear algebra operator nodes, `SplatNode` and `MaxNode`. The `SplatNode` first fills a Tensor with zero, and `MaxNode` compare `Input` with the filling Tensor. We can fuse these two operations which work with the same shape of tensors into a single kernel.
+
+Please refer to the document in [Backend](https://github.com/pytorch/glow/blob/master/docs/Backends.md#backend-specific-nodes-and-instructions) part for source code details on adding a new backend-specific CPUMaxSplatNode on CPU. 
+
+#### Data Layout Transformation for Conv Operator in OpenCL
+
+OpenCL Conv is faster in layout `NCHW` if group > 1. But  the default layout of convolution operator in Glow is `NHWC`. So we transpose the inputs/output and replace the Convolution with a backend-specific `OCLConvolution` that uses `NCHW`. The tansposes mostly can get optimized away thanks to the high-level graph optimizations.
```

Comment:
"OpenCL Conv is faster in layout `NCHW` if group > 1" --- is that accurate? Are you basing that on the [comment from the code](https://github.com/pytorch/glow/blob/master/lib/Backends/OpenCL/Transforms.cpp#L39)? Because I thought the reason we don't use `NCHW` if `group > 1` is because there was some bug in the implementation. I think this shouldn't mention group convolution at all. I think it should say that OpenCL Conv is faster when in `NCHW`, but Glow's default layout is `NHWC`. So when we see a Convolution that we can use `NCHW` ... etc.

Reply:
Got it! Thanks for the detailed interpretation.